Ginger/GingerCoreNET/Database/DatabaseOperations.cs

```diff
@@ -469,65 +471,57 @@ public List<string> GetTablesList(string Keyspace = null)
-                        NoSqlDriver = new GingerCouchbase(Database);
-                        rc = NoSqlDriver.GetTableList(Keyspace);
-                    }
-                    else if (Database.DBType == Database.eDBTypes.MongoDb)
-                    {
-                        NoSqlBase.NoSqlBase NoSqlDriver = null;
-                        NoSqlDriver = new GingerMongoDb(Database);
-                        rc = NoSqlDriver.GetTableList(Keyspace);
-                    }
-                    else if (Database.DBType == eDBTypes.CosmosDb)
-                    {
-                        GingerCosmos objGingerCosmos = new GingerCosmos();
-                        Database.ConnectionString = GetConnectionString();
-                        objGingerCosmos.Db = Database;
-                        rc = objGingerCosmos.GetTableList(Keyspace);
+                        case eDBTypes.Cassandra:
+                            NoSqlBase.NoSqlBase NoSqlDriver = null;
+                            NoSqlDriver = new GingerCassandra(Database);
+                            rc = NoSqlDriver.GetTableList(Keyspace);
+                        break;
+
+                        case eDBTypes.Couchbase:
+                            NoSqlDriver = new GingerCouchbase(Database);
+                            rc = NoSqlDriver.GetTableList(Keyspace);
+                        break;
+
+                        case eDBTypes.MongoDb:
+                            NoSqlDriver = new GingerMongoDb(Database);
+                            rc = NoSqlDriver.GetTableList(Keyspace);
+                        break;
+
+                        case eDBTypes.CosmosDb:
+                            GingerCosmos objGingerCosmos = new GingerCosmos();
+                            Database.ConnectionString = GetConnectionString();
+                            objGingerCosmos.Db = Database;
+                            rc = objGingerCosmos.GetTableList(Keyspace);
+                        break;
+
+                        case eDBTypes.Oracle:
+                            string[] restr = new string[1];
+                            restr[0] = GetConnectedUsername();
+                            DataTable table = await oConn.GetSchemaAsync("Tables", restr);
+                            string tableName = "";
+                            foreach (DataRow row in table.Rows)
+                            {
+                                tableName = (string)row[1];
+                                rc.Add(tableName);
+                            }
+                            break;
+                        default:
+                            //Not implemented
+                            break;
                     }
-                    else
-                    {
+                        if (Database.DBType == eDBTypes.MSSQL || Database.DBType == eDBTypes.MySQL || Database.DBType == eDBTypes.MSAccess || Database.DBType == eDBTypes.DB2 || Database.DBType == eDBTypes.PostgreSQL)
```

Comment:
handle this condition in switch defined above. You can do it using the Case without break.

Reply:
fixed